### PR TITLE
CPP: Add test coverage for LeapYearFieldAccess.isModifiedByArithmeticOperation.

### PR DESCRIPTION
**Pull request by geoffw0**
Add test coverage for `LeapYearFieldAccess.isModifiedByArithmeticOperation()`, which previously could be replaced by `any()` without causing any test failures.  I've also convinced myself that my concerns at https://github.com/Semmle/ql/pull/1354#discussion_r293757113 (never replied to) were unfounded.

### CODE REVIEW

**Comment by geoffw0** [comment]
I've added another test showing that Adding365daysPerYear.ql doesn't quite work as advertised re: requiring the time conversion function being unchecked.  I'm inclined not to worry about fixing this unless/until we see false positives in the wild, as currently the query does not produce many results total so FPs will be very rare.

File: cpp/ql/test/query-tests/Likely Bugs/Leap Year/UncheckedLeapYearAfterYearModification/test.cpp [hunk 1]
```diff
@@ -680,3 +680,57 @@ void mkDateTest(int year)
...
+{
+	SYSTEMTIME st;
+	FILETIME ft;
+	WORD *w_ptr;
+
+	GetSystemTime(&st);
+
+	st.wYear++;
+	st.wYear++;
+	st.wYear++;
```
Comment (jbj) [inline cpp/ql/test/query-tests/Likely Bugs/Leap Year/UncheckedLeapYearAfterYearModification/test.cpp]: These three lines are `BAD` according to `UncheckedLeapYearAfterYearModification.expected`. Please annotate these lines as well as the three lines earlier in this file that are relevant to this query.

File: cpp/ql/test/query-tests/Likely Bugs/Leap Year/Adding365DaysPerYear/test.cpp [hunk 2]
```diff
@@ -197,3 +197,29 @@ time_t mkTime(int days)
...
+	ULONGLONG qwLongTime;
+	qwLongTime = (((ULONGLONG)ft.dwHighDateTime) << 32) + ft.dwLowDateTime;
+
+	// add a year by calculating the ticks in 365 days
+	// (which may be incorrect when crossing a leap day)
+	qwLongTime += 365 * 24 * 60 * 60 * 10000000LLU;
+
+	// copy back to a FILETIME
+	ft.dwLowDateTime = (DWORD)(qwLongTime & 0xFFFFFFFF); // GOOD [FALSE POSITIVE]
+	ft.dwHighDateTime = (DWORD)(qwLongTime >> 32);
```
[the remark this answers is not in the document]
Comment (geoffw0) [reply cpp/ql/test/query-tests/Likely Bugs/Leap Year/Adding365DaysPerYear/test.cpp]: Done.

File: cpp/ql/test/query-tests/Likely Bugs/Leap Year/UncheckedLeapYearAfterYearModification/test.cpp (hunk 1 above)
Comment (geoffw0) [reply cpp/ql/test/query-tests/Likely Bugs/Leap Year/UncheckedLeapYearAfterYearModification/test.cpp]: Done.

**Pull request merged**